lib/SPIRV/SPIRVWriter.cpp

```diff
@@ -1365,14 +1365,31 @@ SPIRVValue *LLVMToSPIRV::transValueWithoutDecoration(Value *V,
     return mapValue(V, BI);
   }
 
-  if (SelectInst *Sel = dyn_cast<SelectInst>(V))
-    return mapValue(
-        V,
-        BM->addSelectInst(
-            transValue(Sel->getCondition(), BB),
-            transValue(Sel->getTrueValue(), BB, true, FuncTransMode::Pointer),
-            transValue(Sel->getFalseValue(), BB, true, FuncTransMode::Pointer),
-            BB));
+  if (SelectInst *Sel = dyn_cast<SelectInst>(V)) {
+    auto CV = transValue(Sel->getCondition(), BB);
+    auto TV = transValue(Sel->getTrueValue(), BB, true, FuncTransMode::Pointer);
+
+    if (!CV->getType()->isTypeVector() && TV->getType()->isTypeVector()) {
+      Type *BoolTy = IntegerType::getInt1Ty(M->getContext());
+      auto VT = transType(FixedVectorType::get(
+          BoolTy, TV->getType()->getVectorComponentCount()));
+      std::vector<SPIRVId> IdVec;
+      for (unsigned Comp = 0; Comp < TV->getType()->getVectorComponentCount();
+           Comp++)
+        IdVec.push_back(CV->getId());
```

Comment:
I guess we can use the fill constructor here:
```suggestion
      std::vector<SPIRVId> IdVec(TV->getType()->getVectorComponentCount(),
                                 CV->getId());
```